pkg/services/publicdashboards/publicdashboard.go

```diff
@@ -12,6 +12,7 @@ import (
 
 // These are the api contracts. The API should match the underlying service and store
 
+//go:generate go run ./commands/generate_datasources/main.go
```

Comment:
I think this should accept an argument for where the generated file should go.

Did you have a chance to test importing the generated file for use in ./service/service.go?

Reply:
Also, we should check-in the initial version of the gen file.

Reply:
Looks like files ending with `_gen.go` are gitignored. So I removed that suffix.

Reply:
For the output file location, would we want it anywhere else besides in the root of the publicdashboards folder?

Reply:
Oh! interesting. I didn't realize _gen.go is ignored. I wonder if that is because they want us to explicitly check that in. Could you do a little extra research on that? I'm sure other gen'd code is checked in. I wonder if there is just a different naming convention or they're preventing accidents.

As long as we can import that package from folders beneath it, that location is fine.

Reply:
Looks like we explicitly exclude certain `*_gen.go` in the gitignore, which makes sense. Added this one to that list and added the _gen suffix back 👍